include/dsn/cpp/auto_codes.h

```diff
@@ -167,56 +167,6 @@ namespace dsn
 #endif
     };
 
-    class atom_int
```

Comment:
I agree, but the reason here is different. If seems that you are going to manually do the marshall/unarshall for app-state etc. ? If we want to do that automatically (e.g., via code generation), we have to add support for atom_int, or we have to eliminate it from app_state.